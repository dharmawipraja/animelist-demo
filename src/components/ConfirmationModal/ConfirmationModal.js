import React from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';

function ConfirmationModal({ isOpen, onClose, onConfirm, title, subtitle }) {

  return (
    <div>
      <Dialog
        open={isOpen}
        onClose={onClose}
        aria-labelledby="confirmation-title"
        aria-describedby="confirmation-subtitle"
      >
        <DialogTitle id="confirmation-title">
          {title}
        </DialogTitle>
        <DialogContent>
          <DialogContentText id="confirmation-subtitle">
            {subtitle}
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={onClose}>Cancel</Button>
          <Button onClick={onConfirm} autoFocus>
            Confirm
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default ConfirmationModal;
