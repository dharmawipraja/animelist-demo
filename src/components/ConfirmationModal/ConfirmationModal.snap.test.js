import renderer from 'react-test-renderer';
import ConfirmationModal from './ConfirmationModal';

jest.mock('@mui/material/Dialog', () => 'Dialog');

describe('ConfirmationModal', () => {
  it('should renders ConfirmationModal component', () => {
    const props = {
      isOpen: true,
      onClose: jest.fn(),
      onConfirm: jest.fn(),
      title: 'Confirmation Modal',
      subtitle: 'This is Confirmation Modal'
    };

    const tree = renderer
      .create(<ConfirmationModal {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
