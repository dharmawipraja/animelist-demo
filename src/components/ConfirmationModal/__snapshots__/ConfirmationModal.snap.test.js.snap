// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal should renders ConfirmationModal component 1`] = `
<div>
  <Dialog
    aria-describedby="confirmation-subtitle"
    aria-labelledby="confirmation-title"
    onClose={[MockFunction]}
    open={true}
  >
    <h2
      className="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
      id="confirmation-title"
    >
      Confirmation Modal
    </h2>
    <div
      className="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
    >
      <p
        className="MuiDialogContentText-root MuiTypography-root MuiTypography-body1 css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        id="confirmation-subtitle"
      >
        This is Confirmation Modal
      </p>
    </div>
    <div
      className="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
    >
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Cancel
      </button>
      <button
        autoFocus={true}
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Confirm
      </button>
    </div>
  </Dialog>
</div>
`;
